doc/glossary.rst

```diff
@@ -0,0 +1,1268 @@
+        sample properties and their routing in :term:`meta-estimators`, though
+        a ``fit_params`` parameter is often used.
+
+    scikit-learn-contrib
+        A venue for publishing Scikit-learn-compatible libraries that are
+        broadly authorized by the core developers and the contrib community,
+        but not maintained by the core developer team.
+        See `http://scikit-learn-contrib.github.io`_
+
+    sparse matrix
+        A representation of two-dimensional numeric data that is more memory
+        efficient the corresponding dense numpy array where almost all elements
+        are zero. We use the :mod:`scipy.sparse` framework, which provides
+        several underlying sparse data representations, or *formats*.
+        Some formats are more efficient than others for particular tasks, and
+        when a particular format provides especial benefit, we try to document
+        this fact in Scikit-learn parameter descriptions.
+
+        Some sparse matrix formats (notably CSR, CSC, COO and LIL) distinguish
+        between *implicit* and *explicit* zeros. Explicit zeros are stored
+        (i.e. they consume memory in a ``data`` array) in the data structure,
+        while implicit zeros correspond to every element not otherwise defined
+        in explicit storage.
+
+        Two semantics for sparse matrices are used in Scikit-learn:
+
+        matrix semantics
+            The sparse matrix is interpreted as an array with implicit and
+            explicit zeros being interpreted as the number 0.  This is the
+            interpretation most often adopted, e.g. when sparse matrices
+            are used for feature matrices or multilabel indicator matrices.
+        graph semantics
+            As with :mod:`scipy.sparse.csgraph`, explicit zeros are
+            interpreted as the number 0, but implicit zeros indicate a masked
+            or absent value, such as the absence of an edge between two
+            vertices of a graph, where an explicit value indicates an edge's
+            weight. This interpretation is adopted to represent connectivity
+            in clustering, in representations of nearest neighborhoods
+            (e.g. :func:`neighbors.kneighbors_graph`), and for precomputed
+            distance representation where only distances in the neighborhood
+            of each point are required.
+
+    target
+    targets
+        TODO
+
+        Dependent variable or outcome variable.
+
+    transduction
+    transductive
+        A transductive (contrasted with :term:`inductive`) machine learning
+        method is designed to model a specific dataset, but not to apply that
+        model to unseen data.  Examples include :class:`manifold.TSNE`,
+        :class:`cluster.AgglomerativeClustering` and
+        :class:`neighbors.LocalOutlierFactor`.
+
+    unlabeled
+    unlabeled data
+        TODO
+
+    ``X``
+        Denotes data that is observed at training and prediction time, used as
```

Comment:
training *or* prediction time? Oh I guess you mean as opposed to ``y``? Well we can iterate on this description ;)